[FAB-17605] Unhandled InitFactories error

Eggsyz opened the pull request:
<!--- DELETE MARKDOWN COMMENTS BEFORE SUBMITTING PULL REQUEST. -->

<!--- Provide a descriptive summary of your changes in the Title above. -->

#### Type of change

<!--- What type of change? Pick one option and delete the others. -->

- Bug fix

#### Description

<!--- Describe your changes in detail, including motivation. -->
A project using func InitFactories, it output errors when build it. But many places do not handle errors.
#### Additional details

<!--- Additional implementation details or comments to reviewers. -->
<!--- Summarize how the pull request was tested (if not obvious from commit). -->

#### Related issues

<!--- Include a link to any associated issues, e.g. Jira issue or approved rfc. -->
[FAB-17605](https://jira.hyperledger.org/browse/FAB-17605?jql=project%20%3D%20FAB)

<!---
#### Release Note
If change impacts current users, uncomment Release Note heading and provide
release note text.
Also, copy release note text into the release specific /release_notes file.
-->

<!--
Checklist (DELETE AFTER READING):

- `Signed-off-by` added to commits (required for DCO check to pass)
- Tests have been added/updated (required for bug fixes and features)
- Unit and/or integration tests pass locally
- Run linters and checks locally using 'make checks'
- If change requires documentation updates, make updates in pull request,
  or open a separate issue and provide link
- Squash commits into a single commit, unless a stack of commits is
  intentional to assist reviewers or to preserve review comments.
- For additional contribution guidelines see the project's CONTRIBUTING.md file
-->

Commit:
Added Jenkinsfile

Commit:
Delete Jenkinsfile

Eggsyz committed:
Merge remote-tracking branch 'github/master'

Eggsyz committed:
[FAB-17605] Unhandled InitFactories() error
A project using func InitFactories, it output errors when build it. But many places do not handle errors.

Signed-off-by: jilg <liguo_ji@outlook.com>

Eggsyz committed:
Merge branch 'master' of https://github.com/hyperledger/fabric into handler

denyeart commented on internal/peer/chaincode/common_test.go:
```diff
@@ -184,8 +188,11 @@ func TestGetOrdererEndpointFail(t *testing.T) {
+	// Init the BCCSP
+	err = factory.InitFactories(nil)
+	if err != nil {
+		assert.Error(t, err, "Could not initialize BCCSP Factories")
```
Error is not expected here, so why assert it?
BTW expected error would go before actual error in assert.Error().

lindluni replied:
Use assert.NoError(t, err) which will flag unexpected introduced behavior

denyeart commented on internal/peer/chaincode/common_test.go:
```diff
@@ -159,7 +159,11 @@ func TestGetOrdererEndpointFromConfigTx(t *testing.T) {
+	// Init the BCCSP
+	err = factory.InitFactories(nil)
+	if err != nil {
+		assert.Error(t, err, "Could not initialize BCCSP Factories")
```
See other comment.

denyeart commented on cmd/configtxgen/main_test.go:
```diff
@@ -177,12 +177,16 @@ func TestBlockFlags(t *testing.T) {
+	// Init the BCCSP
+	err := factory.InitFactories(nil)
+	if err != nil {
+		assert.Error(t, err, "Could not initialize BCCSP Factories")
```
See other comment.

Eggsyz replied:
thank you. I will fix it

denyeart commented on cmd/configtxgen/main.go:
```diff
@@ -255,7 +255,11 @@ func main() {
+	// Init the BCCSP
+	err := factory.InitFactories(nil)
+	if err != nil {
+		logger.Fatalf("Could not initialize BCCSP Factories [%s]", err)
```
Why does this one log it, but the other ones panic?

Eggsyz committed:
[FAB-17605] Unhandled InitFactories() error
    A project using func InitFactories, it output errors when build it. But many places do not handle errors.

Signed-off-by: jilg <liguo_ji@outlook.com>

Eggsyz committed:
Merge branch 'master' of https://github.com/Eggsyz/fabric into eggsy_master

Eggsyz committed:
Merge branch 'eggsy_master' into handler
